pyannote/audio/tasks/segmentation/unsupervised_segmentation.py

```diff
@@ -0,0 +1,153 @@
+from typing import List, Text, Tuple, Union
+
+import numpy as np
+import torch
+from pyannote.core import Segment, SlidingWindowFeature
+from pyannote.database import Protocol
+from torch_audiomentations.core.transforms_interface import BaseWaveformTransform
+from typing_extensions import Literal
+
+from pyannote.audio.core.io import AudioFile
+from pyannote.audio.core.model import Model
+from pyannote.audio.core.task import Task
+from pyannote.audio.tasks import Segmentation
+
+
+class UnsupervisedSegmentation(Segmentation, Task):
+    def __init__(
+        self,
+        model: Model,  # unsupervised param: model to use to generate truth
+        protocol: Protocol,
+        fake_in_train=True,  # generate fake truth in training mode
+        fake_in_val=True,  # generate fake truth in val mode
+        # supervised params
+        duration: float = 2.0,
+        max_num_speakers: int = None,
+        warm_up: Union[float, Tuple[float, float]] = 0.0,
+        overlap: dict = Segmentation.OVERLAP_DEFAULTS,
+        balance: Text = None,
+        weight: Text = None,
+        batch_size: int = 32,
+        num_workers: int = None,
+        pin_memory: bool = False,
+        augmentation: BaseWaveformTransform = None,
+        loss: Literal["bce", "mse"] = "bce",
+        vad_loss: Literal["bce", "mse"] = None,
+    ):
+        super().__init__(
+            # Mixin params
+            protocol,
+            duration=duration,
+            warm_up=warm_up,
+            batch_size=batch_size,
+            num_workers=num_workers,
+            pin_memory=pin_memory,
+            augmentation=augmentation,
+            # Segmentation params
+            max_num_speakers=max_num_speakers,
+            overlap=overlap,
+            balance=balance,
+            weight=weight,
+            loss=loss,
+            vad_loss=vad_loss,
+        )
+
+        self.m0 = model
+        self.fake_in_train = fake_in_train
+        self.fake_in_val = fake_in_val
+
+    def get_truth_from_model(
+        self, waveforms: torch.Tensor, model: Model = None
+    ) -> Tuple[np.ndarray, List[Text]]:
+        if model is None:
+            model = self.m0
+
+        y = self.m0(waveforms=waveforms)[0, :, :]
+
+        # dirty float output to int output
+        y = torch.round(y)
+        y = y.type(torch.int8)
```

Comment:
This seems like a good initial guess (equivalent to comparing to a 0.5 threshold).
It is not clear yet how we could improve the value of this threshold. 
Yet another research problem to explore :-) Maybe open a dedicated issue in your repo so that can keep track of stuff to investigate?